# HG changeset patch
# User Carnë Draug <carandraug@octave.org>
# Date 1382554110 -3600
#      Wed Oct 23 19:48:30 2013 +0100
# Node ID 32344942ee6eee5a9678155d75f61e69a6a7fa4a
# Parent  c6891317ff27cfa735c64955216cfca5a9644999
Distribute an AppData for Octave as suggested by freedesktop.org (bug #40356)

* etc/octave.appdata.ml: AppData file for Octave.
* Makefile.am: add AppData to EXTRA_DIST.

diff --git a/Makefile.am b/Makefile.am
--- a/Makefile.am
+++ b/Makefile.am
@@ -59,16 +59,17 @@ EXTRA_DIST = \
   etc/README.Cygwin \
   etc/README.Linux \
   etc/README.MacOS \
   etc/README.MinGW \
   etc/README.Windows \
   etc/README.gnuplot \
   etc/README.kpathsea \
   etc/gdbinit \
+  etc/octave.appdata.xml \
   run-octave.in
 
 include m4/module.mk
 
 # Subdirectories in which to run `make all'.  Including "." before
 # @DOCDIR@ is an attempt to force all preceding directories in the list
 # to be processed before the current directory so that the
 # scripts/DOCSTRINGS libinterp/DOCSTRINGS files are built before
diff --git a/etc/octave.appdata.xml b/etc/octave.appdata.xml
new file mode 100644
--- /dev/null
+++ b/etc/octave.appdata.xml
@@ -0,0 +1,56 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<!--
+  Copyright (C) 2013 Carnë Draug
+
+  This program is free software: you can redistribute it and/or modify
+  it under the terms of the GNU General Public License as published by
+  the Free Software Foundation, either version 3 of the License, or
+  (at your option) any later version.
+
+  This program is distributed in the hope that it will be useful,
+  but WITHOUT ANY WARRANTY; without even the implied warranty of
+  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+  GNU General Public License for more details.
+
+  You should have received a copy of the GNU General Public License
+  along with this program.  If not, see <http://www.gnu.org/licenses/>.
+-->
+<application>
+  <id type="desktop">octave.desktop</id>
+  <licence>GPLv3+</licence>
+  <name>GNU Octave</name>
+  <summary>Interactive programming environment for numerical computations</summary>
+  <description>
+    <p>
+      GNU Octave is a high-level interpreted language, primarily intended for
+      numerical computations. It provides capabilities for the numerical
+      solution of linear and nonlinear problems, and for performing other
+      numerical experiments. It also provides extensive graphics capabilities
+      for data visualization and manipulation. Octave is normally used through
+      its interactive command line interface, but it can also be used to write
+      non-interactive programs. The Octave language is quite similar to
+      Matlab so that most programs are easily portable.
+    </p>
+    <p>
+      Octave has extensive tools for solving common numerical linear algebra
+      problems, finding the roots of nonlinear equations, integrating ordinary
+      functions, manipulating polynomials, and integrating ordinary
+      differential and differential-algebraic equations. It is easily
+      extensible and customizable via user-defined functions written in
+      Octave's own language, or using dynamically loaded modules written in
+      C++, C, Fortran, or other languages.
+    </p>
+  </description>
+  <screenshots>
+    <!--
+      From: http://people.freedesktop.org/~hughsient/appdata/
+      All screenshots should have a 16:9 aspect ratio, and should have
+      a width that is no smaller than 620px. Screenshots should be in PNG
+      or JPEG format. PNG is the preferred format.
+    -->
+    <screenshot type="default">http://www.gnu.org/software/octave/images/screenshot.png</screenshot>
+  </screenshots>
+  <url type="homepage">http://www.octave.org</url>
+  <updatecontact>maintainers@octave.org</updatecontact>
+  <project_group>GNU</project_group>
+</application>
